00_ES6/00_start/src/pages/chat/chat.container.jsx

```diff
@@ -1,19 +1,81 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
+import { messageFactory } from '../../api/chat'
 import { withSessionContext } from '../../common';
+import { ChatComponent } from './chat.component';
+import {establishRoomSocketConnection, messagesToString} from './chat.container.business'
 
-const ChatContainerInner = (props) =>
-  <React.Fragment>
-    <h1>Hello from chat page</h1>
-    <p>nickname: {props.sessionInfo.nickname}</p>
-    <p>room: {props.sessionInfo.room}</p>
-    <Link to="/">Navigate back to lobby</Link>
-  </React.Fragment>
+export class ChatContainerInner extends React.Component {
+  
+  constructor(props) {
+    super(props);
+
+    this.state = {currentMessage: '', chatLog: ''};    
+    this.socket = null;
+    this.messageFactory = null;
+  }
+  
+  enrollRoom = () => {
+    this.socket = establishRoomSocketConnection(this.props.sessionInfo.nickname, this.props.sessionInfo.room);
+    this.messageFactory = messageFactory(this.props.sessionInfo.room, this.props.sessionInfo.nickname);
+    this.setupSocketListeners(this.socket);
+  }
+
+  disconnectfromRoom = () => {
+    this.socket.disconnect();
+  }
+
+  setupSocketListeners(socket) {
+    socket.on('connect', () => {
+      console.log(socket.id);
+      socket.emit('messages');
+    });
+    socket.on('error', (err) => console.log(err));
+    socket.on('disconnect', () => console.log('disconnected'))
+
+    socket.on('message', (msg) => {
+      this.setState({chatLog: `${this.state.chatLog}${msg.user}: ${msg.text}\n`});
+      console.log(msg);
+    });
+    socket.on('messages', (msgs) => {
+      let messages = messagesToString(msgs);
+
+      this.setState({chatLog: `${this.state.chatLog}${messages}`});      
+    });
+  }
```

Comment:
Since there are a lot of callbacks and setState I think this is a perfect place to show later on the power of sagas.